static/js/models/thing-model.js

```diff
@@ -202,10 +202,10 @@ class ThingModel extends Model {
       this.base
     );
 
-    return API.putJson(href, value)
-      .then((json) => {
+    return API.putJsonWithEmptyResponse(href, value)
+      .then(() => {
         const result = {};
-        result[name] = json;
+        result[name] = value;
```

Comment:
I think it is worth adding a comment in the function documentation about the fact that the returned result is just for API backward compatibility and it might not reflect the actual remote property value.